app/src/main/java/com/mapbox/maps/testapp/examples/LargeGeojsonPerformanceActivity.kt

```diff
@@ -0,0 +1,139 @@
+package com.mapbox.maps.testapp.examples
+
+import android.os.Bundle
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.geojson.Feature
+import com.mapbox.geojson.FeatureCollection
+import com.mapbox.geojson.LineString
+import com.mapbox.geojson.Point
+import com.mapbox.maps.CameraOptions
+import com.mapbox.maps.Style
+import com.mapbox.maps.extension.style.layers.addLayer
+import com.mapbox.maps.extension.style.layers.generated.lineLayer
+import com.mapbox.maps.extension.style.sources.addSource
+import com.mapbox.maps.extension.style.sources.generated.geoJsonSource
+import com.mapbox.maps.plugin.animation.MapAnimationOptions
+import com.mapbox.maps.plugin.animation.flyTo
+import com.mapbox.maps.testapp.R
+import com.mapbox.maps.testapp.examples.annotation.AnnotationUtils
+import kotlinx.android.synthetic.main.activity_large_geojson.*
+
+/**
+ * Example of passing large geojson to verify it does not block UI thread.
+ */
+class LargeGeojsonPerformanceActivity : AppCompatActivity() {
+
+  private lateinit var routePoints: FeatureCollection
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_large_geojson)
+
+    routePoints = FeatureCollection.fromFeature(
+      Feature.fromGeometry(
+        LineString.fromPolyline(
+          DirectionsResponse.fromJson(
+            AnnotationUtils.loadStringFromAssets(
+              this@LargeGeojsonPerformanceActivity, "long_route.json"
+            )
+          ).routes()[0].geometry()!!,
+          6
+        )
+      )
+    )
+
+    mapView.getMapboxMap()
+      .apply {
+        setCamera(
+          CameraOptions.Builder()
+            .center(Point.fromLngLat(LONGITUDE, LATITUDE))
+            .zoom(START_ZOOM)
+            .build()
+        )
+        loadStyleUri(Style.MAPBOX_STREETS) { style ->
```

Comment:
We have the default style now, can we change to `getStyle` here?

Reply:
This code block is inside the onCreate, as far as I remember, the default style is loaded in the onStart? will the added RuntimeStyle be overwritten by the style?

Reply:
No, we have check there, if one style is already set we will not set the default style.